scripts/validate_string_concatenation.py

```diff
@@ -0,0 +1,148 @@
+    """
+    Main entry point of the script.
+
+    Parameters
+    ----------
+    source_path : str, default '.'
+        Source path representing path to a file/directory.
+    output_format : {'default', 'azure'}, default 'default'
+        Output format of the script.
+
+    Returns
+    -------
+    bool
+        True if found any strings that needs to be concatenated.
+
+    Raises
+    ------
+    ValueError
+        If the `source_path` is not pointing to existing file/directory.
+    """
+    if not os.path.exists(source_path):
+        raise ValueError(
+            "Please enter a valid path, pointing to a valid file/directory."
+        )
+
+    is_failed: bool = False
+    msg: str = OUTPUT_FORMATS.get(output_format)
+
+    if os.path.isfile(source_path):
+        for values in strings_to_concatenate(source_path):
+            is_failed = True
+            print(msg.format(**values))
+
+    for subdir, _, files in os.walk(source_path):
+        for file_name in files:
+            if any(
+                file_name.endswith(extension) for extension in FILE_EXTENSIONS_TO_CHECK
+            ):
+                for values in strings_to_concatenate(os.path.join(subdir, file_name)):
+                    is_failed = True
+                    print(msg.format(**values))
+    return is_failed
+
+
+def strings_to_concatenate(source_path: str) -> Generator[Dict[str, str], None, None]:
+    """
+    Yielding the strings that needs to be concatenated in a given file.
+
+    Parameters
+    ----------
+    source_path : str
+        File path pointing to a single file.
+
+    Yields
+    ------
+    Dict[str, str]
+        Containing:
+            source_path
+                Source file path.
+            line_number
+                Line number of unconcatenated string.
+            start
+                Starting string of unconcatenated string.
+            end
+                Ending string of unconcatenated string.
+    """
+    with open(source_path, "r") as file_name:
+        tokens: List = list(tokenize.generate_tokens(file_name.readline))
+
+    for current_token, next_token in zip(tokens, tokens[1:]):
+        if current_token[0] == next_token[0] == token.STRING:
+            yield {
+                "source_path": source_path,
+                "line_number": current_token[2][0],
+                "start": current_token[1],
+                "end": next_token[1],
+            }
+
+
+if __name__ == "__main__":
+    argparser = argparse.ArgumentParser(description="Validate concatenated strings")
+    argparser.add_argument(
+        "--format",
+        "-f",
+        default="default",
+        choices=AVAILABLE_OUTPUT_FORMATS,
+        help=(
+            f"Format of the output when validating strings "
+            f"it can be {AVAILABLE_OUTPUT_FORMATS}."
+        ),
+    )
+
+    argparser.add_argument(
+        "--path", "-p", default=".", help=f"Source path of file/directory to check.",
```

Comment:
make this a positional argument please, so instead of `script -p path` we simply call `script path`